### ISSUE TITLE
[FR] Add 'delete' and 'add to comparison' buttons in the run view page

### ISSUE THREAD

**Issue by miwojc** [mlflow/mlflow#2184]
## Describe the proposal
Add 'delete' and 'add to comparison' buttons in the run view page.

### Motivation
When on run view page, it would be much easier to delete the run from that page, rather than having to go back to the runs list page and find that run again and then delete the run. Similarly for run comparisons.

### Proposed Changes
Add 'delete' and 'add to comparison' buttons in the run view page.

**Comment by mparkhe** [comment]
Delete button makes sense. However, "add to comparison" seems complex and could be confusing--specifically it is harder for a user to navigate a run's page and then adding it comparisons list--if it can done from runs list page.

What do you think: @gioa and @Zangr.

**Comment by gioa** [comment]
+1 on adding a Delete button on the run view page. 

For "add to comparison", some E-commence sites add a comparison section whenever users click on "compare". 
![BestBuyComparison](https://user-images.githubusercontent.com/1960721/70655913-ee83f780-1c0d-11ea-97ea-69c7a17f8f06.png)

Another way is to allow add & remove runs in the comparison page. So clicking on an "add to comparison" in the run view page can direct users to this comparison page with only one item displayed, users need to add more runs on the top field to compare though. 

![MLflowComparison](https://user-images.githubusercontent.com/1960721/70656171-67834f00-1c0e-11ea-9fad-ec4a180f6a9a.png)

What do you think?

**Comment by Yuchen-Wang-SH** [comment]
Hi @gioa and @mparkhe ! I am new to the MLflow community but I think I can add this "delete" button to the run page as my first issue here. Do you still need this FR?

**Comment by gioa** [comment]
@Yuchen-Wang-SH Yes, that would be nice! Could you add the delete action in the same dropdown as the "Rename" on the run detail page (see attached), and show a confirmation dialog (similar to deleting a run in the experiment page)?

Let me know if you have any questions! CC @Zangr 


![mock](https://user-images.githubusercontent.com/1960721/72294184-dedf4080-3609-11ea-82ae-20797488aae6.png)

**Comment by Yuchen-Wang-SH** [comment]
@gioa Hi Joy. I am wondering what the desired behavior should be after deleting a run on its own page. It seems like deleted run pages can still be accessed by design, and the page seems identical to me no matter whether this run has been deleted or not. Would you please give more instructions on this?

**Comment by gioa** [comment]
@Yuchen-Wang-SH That's a good point! We don't do anything today but ideally, we should block users from any further actions on a deleted run like rename, add notes, edit tags, or register model. We could also show a visual banner so users can differentiate it easily. 

See the attached mock(added a visual banner and removed all the action buttons). What do you think? Will that be something you'd interested in adding? 

![MLFlow_RunDetail](https://user-images.githubusercontent.com/1960721/72395586-d7459780-36ee-11ea-9b91-e49d16fb3c33.png)

**Comment by Yuchen-Wang-SH** [comment]
@gioa Hi Joy. I have implemented most of the features mentioned. Few things here:
1. What is the React Component Name of the "This run is deleted..." banner on your mock, if you have implemented something like that before?
2. I can disable the note editing by not allowing the icon shows up on a deleted run page. But to disable tag editing and model registering it seems like I have to add a `canEdit` prop to the respective React component, and let that component decide whether to display the editing field or the editing button. Please tell me whether changing the props of your React components is acceptable. @Zangr 
Thanks!

**Comment by gioa** [comment]
@Yuchen-Wang-SH Thanks for working on it! I think @Zangr used the [alert component](https://ant.design/components/alert/) from Ant Design and customized the style a little bit in mlflow/server/js/src/model-registry/components/ModelVersionView.css.

**Comment by Yuchen-Wang-SH** [comment]
@gioa Hi Joy. I have opened this pull request #2310 . I have implemented everything you mentioned above except:

1. Putting a `restore` link in the ant design alert component. I am not sure whether we can do this since we are passing a bare string as property to this component.
2. Disabling tag editing and model registering for deleted run. Implementing this may lead to an additional `canEdit` prop to these two components, I am not sure whether you are okay with this. And it seems like there is already some logic stopping you from doing these changing.

Thank you!
